src/nvim/menu.c

```diff
@@ -1156,14 +1154,18 @@ static char_u *popup_mode_name(char_u *name, int idx)
 }
 
 
-/*
- * Duplicate the menu item text and then process to see if a mnemonic key
- * and/or accelerator text has been identified.
- * Returns a pointer to allocated memory, or NULL for failure.
- * If mnemonic != NULL, *mnemonic is set to the character after the first '&'.
- * If actext != NULL, *actext is set to the text after the first TAB.
- */
+/// Duplicate the menu item text and then process to see if a mnemonic key
+/// and/or accelerator text has been identified.
+///
+/// @param str The menu item text.
+/// @param[out] mnemonic If non-NULL, *mnemonic is set to the character after
+///             the first '&'.
+/// @param[out] actext If non-NULL, *actext is set to the text after the first
+///             TAB.
+///
+/// @return a pointer to allocated memory.
 static char_u *menu_text(char_u *str, int *mnemonic, char_u **actext)
+  FUNC_ATTR_NONNULL_RET FUNC_ATTR_WARN_UNUSED_RESULT
```

Comment:
Thanks for the input, I've updated the PR. But I think it would be more efficient to prepare special commit(s) which deal with function attributes, constness, etc., otherwise the commit essense is buried under other refactorings.